### DISCUSSION TITLE
Allow 'layout' declaration in markdown front matter to resolve aliases

### DISCUSSION THREAD

**Discussion by jscheller** [withastro/roadmap#76] [Proposal]
To render markdown files we need to include a layout file in the markdown file's front matter...

```
layout: ../layouts/BaseLayout.astro
```

...but this is an absolute path that is relative to the markdown file being rendered. If a project has a nested directory structure full of markdown files in the project's "pages" directory, then the required layout declaration might need to be arbitrarily set for different files, and then updated if those files are moved around around. For example...

```
  /src
     /layouts
         BaseLayout.astro
     /pages
         foo.md
         /bar
            fee.md
            /baz
               fi.md
               fo.md
```

In this example, the file foo.md would need to reference the layout file as "../layouts/BaseLayout.astro", where as fee.md would reference it as "../../layouts/BaseLayout.astro" and both fi.md and fo.md would need "../../../layouts/BaseLayout.astro". All of this is quite brittle even if it were initially setup properly.

As a solution, it would be desirable for the layout property in the markdown front matter to resolve aliases declared in the Astro configuration, much in the way that import statements for markdown components allow for aliases to be resolved. In this case, you would add something like this to the tsconfig.json file...

```   
 "paths": {
      "@/layouts/*": ["src/layouts/*"]
   }
```

...and then in the layout files, use this to reference the layout file...

```
layout: @/layouts/BaseLayout.astro
```

...regardless of the markdown file's relative nesting in the file system.

**Comment by FredKSchott** [comment]
+1! I think this could even be considered a bug, since I can't think of a reason why the rest of Astro supports aliases but this one property does not.

**Comment by flo8** [comment]
I was also looking for a way to avoid to end up with "../../../layouts/BaseLayout.astro" kind of path. Our .md files are also on different levels so we have to make sure the number of ../../../ is correct for each level. It's time consuming and also error prone.

I tried to use absolute path but can't find anything in the documentation? Even without aliases, is there a way to use absolute path from the ./src?

Many thanks

**Reply by delucis** [reply]
Aliases are supported for Markdown layouts now and are probably the best approach for this I think. There’s an example of how to set up aliases in the docs here: https://docs.astro.build/en/guides/aliases/

**Reply by flo8** [reply]
Thank you. I didn't notice this feature, works great indeed!

**Comment by louiss0** [comment]
I created a plugin that allows you create layouts for every page in a folder so that you don't ever have to use the layout file for every file that you write. https://www.npmjs.com/package/remark-astro-layout-url-injector
